src/mk/def_rules.mk

```diff
@@ -890,6 +966,18 @@ define SQLC_TO_MODULE_RECIPE =
 	@rm "$<-1252"
 endef
 
+# Temp: Convert UTF-8 to temporary Windows Latin-1, because SQLC pre-compiler doesn't understand UTF-8
+define SQLCPP_TO_MODULE_RECIPE = 
+	$(MODULE_VARIABLES)
+	@$(call echo_cmd,"=== Creating SQLCPP module [$(notdir $<)]")
+    @qsh_out -c "touch -C 1252 $<-1252 && cat $< >$<-1252"
+	$(eval crtcmd := crtsqlcppi obj($(OBJLIB)/$(basename $(@F))) srcstmf('$<-1252') $(CRTSQLCPPIFLAGS))
+	@$(PRESETUP) \
+	$(SCRIPTSPATH)/launch "$(JOBLOGFILE)" "$(crtcmd)" >> $(LOGFILE) 2>&1 && $(call logSuccess,$@) || $(call logFail,$@)
+	@($(call EVFEVENT_DOWNLOAD,$(basename $(@F)).evfevent); ret=$$?; rm $(DEPDIR)/$*.Td 2>/dev/null; rm "$<-1252" 2>/dev/null; exit $$ret)
+	@rm "$<-1252"
```

Comment:
I see this file being deleted twice.  What is this file for?
So does this work.  I.e. the CPP front end is not a PASE executable that is called from an ILE command

Reply:
Ah, it should only be deleted once. Good catch. As for the file, I replicated code from the CRTSQLCI path that (I presume) fixed a problem dealing with CCSID and the SQL precompiler.  Perhaps this is code that should be revisited?  Here is the comment from the code that I replicated:  # Temp: Convert UTF-8 to temporary Windows Latin-1, because SQLC pre-compiler doesn't understand UTF-8

Reply:
Right, the proper solution is to specify TGTCCSID on the compiler so that it converts to a target EBCDIC CCSID before compiling.
This is supported by RPGLE, CLLE, and CBLLE
But I don't know if CRTSQLCI supports it
1252 might not work for all languages, seems fragile to me

Reply:
It looks like CRTSQLCI and CRTSQLCPPI both have this parm :     Conversion CCSID (CVTCCSID) - Help             
                                                             
  The CCSID used for conversion when a UTF-8 source stream   
  file is provided on the Source stream file (SRCSTMF)       
  parameter.  Otherwise, this parameter is ignored.

Reply:
This is how RPG solves it https://www.ibm.com/support/pages/compile-rpg-unicode-source-new-tgtccsid-parameter-71-72-73

Reply:
Yes CVTCCSID is perfect
SO the value is derived from .ibmi.json
and is stored in TGTCCSID variable